Add substrait feature

AntoinePrv opened the pull request:
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

AntoinePrv committed:
Add substrait feature

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

AntoinePrv commented:
@conda-forge/datafusion What do you think of adding this feature? (I'm interested in it)

jakirkham commented:
Friendly nudge @conda-forge/datafusion

(guessing this understandably got lost during the holidays)

AntoinePrv commented:
Last PR I submitted on this feedstock was eventually merged by @xhochy.

AntoinePrv commented:
@conda-forge/datafusion @conda-forge-admin any update on this?

xhochy commented:
@AntoinePrv Can you please add yourself as a maintainer?

AntoinePrv committed:
Add AntoinePrv as maintainer

AntoinePrv committed:
MNT: Re-rendered with conda-smithy 3.54.1 and conda-forge-pinning 2026.01.26.03.13.43

Other tools:
- conda-build 25.11.1
- rattler-build 0.55.1
- rattler-build-conda-compat 1.4.10

AntoinePrv commented:
This is new:
> unknown relocation (31) against symbol ring_core_0_17_14__limbs_mul_add_limb

AntoinePrv committed:
Try override linker

AntoinePrv committed:
Try override linker
